pychromecast/discovery.py

```diff
@@ -1,68 +1,120 @@
 """Discovers Chromecasts on the network using mDNS/zeroconf."""
+import abc
+from collections import namedtuple
+import itertools
 import logging
-import socket
-from threading import Event
+import threading
+import time
 from uuid import UUID
 
 import zeroconf
 
+from .const import SERVICE_TYPE_HOST, SERVICE_TYPE_MDNS
+from .dial import get_device_status, get_multizone_status
+
 DISCOVER_TIMEOUT = 5
 
+ServiceInfo = namedtuple("ServiceInfo", ["type", "data"])
+CastInfo = namedtuple(
```

Comment:
Are we ready yet to migrate to dataclasses?

Reply:
I guess the downside for `ServiceInfo` is that it won't be hashable unless it's frozen. Guess not so useful besides some typing help.

Reply:
[You can construct `namedtuple`s with type annotations like you can with `dataclasses` by subclassing  `typing.NamedTuple`](https://docs.python.org/3/library/typing.html#typing.NamedTuple): 
```python3
from typing import NamedTuple, Optional

class ServiceInfo(NamedTuple):
  type: str
  data: Optional[str] = None

example = ServiceInfo('example')
```

Reply:
@alexdelorenzo There are no type annotations in pychromecast yet, would there be any benefit of using this?

Reply:
@emontnemery I bring it up because of @balloob's comment about moving to dataclasses and typing. There's no real benefit other than typing for namedtuples.

Reply:
OK, thanks!